### ISSUE TITLE
Check if we can use integrated OIDN for viewport renders

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/BlendLuxCore#508]
This would require the use of OIDN's progress monitor callback feature, which is described here: https://github.com/OpenImageDenoise/oidn#filter

> Filters support a progress monitor callback mechanism that can be used to report progress of filter operations and to cancel them as well.

The benefit of using the integrated OIDN would be that packaging of the addon would become simpler and more future-proof because we would no longer rely on OIDN's standalone example executable to do viewport denoising.
However, for this to work, it needs to be possible for the addon to interrupt the denoising process fast and at any time.

**Comment by Dade916** [comment]
Note: https://twitter.com/attila_afra/status/1281567944296865793?s=19

We are using v1.2.1

**Comment by Theverat** [comment]
Yes, that's how I stumbled upon the feature with the progress callback.
I guess we'll update to 1.2.2 after the v2.4 release and then this issue can be implemented.

**Comment by howetuft** [comment]
Yes, I agree, the use of external OIDN instead of integrated one is clumsy, redundant and bug-prone.
